components/centraldashboard/Dockerfile

```diff
@@ -1,20 +1,13 @@
-FROM golang:1.10 as builder
+FROM node:10
 
-# Download and install the latest release of dep
-ADD https://github.com/golang/dep/releases/download/v0.4.1/dep-linux-amd64 /usr/bin/dep
-RUN chmod +x /usr/bin/dep
+WORKDIR /centraldashboard
 
-# Copy the code from the host and compile it
-WORKDIR $GOPATH/src/github.com/kubeflow/kubeflow/components/centraldashboard
-COPY Gopkg.toml Gopkg.lock ./
-RUN dep ensure --vendor-only
-COPY . ./
-RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix nocgo -o centraldashboard .
+COPY app/ ./app
+COPY public/ ./public
+COPY webpack.config.js *.json ./
 
-FROM alpine
-RUN apk --no-cache add ca-certificates
-WORKDIR /app
-COPY --from=builder /go/src/github.com/kubeflow/kubeflow/components/centraldashboard/ /app
-ENV PORT_1=8082
-EXPOSE 8082
-CMD [ "/app/centraldashboard" ]
+RUN npm install && npm run build
```

Comment:
Is there a reason to leave source in the image, when we already have the bundles?

Reply:
If your concern is the size of the build, I added `npm prune --production` to remove any of the devDependencies in node_modules/. That trims about 100MB from the image size. The base image for node:10 is about 900MB so we're not trimming a ton. We can optimize a bit more if we need to shrink this further.

Reply:
We can look into it if it seems to be a concern. < 1GB is not bad